jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -1830,7 +1830,8 @@ void getBytes(int start, int end, char[] data, int index) {
 	 *				when buffer is null
 	 */
 	public void getChars(int start, int end, char[] data, int index) {
-		if (0 <= start && start <= end && end <= lengthInternal()) {
+		// {@code data.length} comes with a null check
+		if (0 <= start && start <= end && end <= lengthInternal() && 0 <= index && end - start <= data.length - index) {
```

Comment:
Some additional brackets would help clarify the grouping here - `((end - start) <= (data.length - index))`